### DISCUSSION TITLE
How to run the tests without the UI ?

### DISCUSSION THREAD

**Discussion by MCord** [fzyzcjy/flutter_convenient_test#254] [Q&A] [answered]
The documentation specifies that the tests can be executed in headless mode but I could not figure out how to achieve that. Could you provide the commands needed to do that ? 

I have found this in one issue but this seems to run the manager.

```
dart run convenient_test_manager_dart # in one shell
flutter run path/to/your/integration_test/main_test.dart --host-vmservice-port 9753 --disable-service-auth-codes --dart-define CONVENIENT_TEST_APP_CODE_DIR=sth # in another shell, run your worker app
```

Ideally I would like to run the tests and just read the log files and optionally generate report files for later inspection in case of failure in my CI pipeline. Is that something supported currently or planned in the feature ? I will be happy to help the project in any way if you could provide some guidance.

Thanks again for the great gift to the community !

**Comment by fzyzcjy** [comment, marked as answer]
You are welcome!

> Ideally I would like to run the tests and just read the log files and optionally generate report files for later inspection in case of failure in my CI pipeline. Is that something supported currently or planned in the feature

Currently supported :) Indeed it runs well in my ci.

> Could you provide the commands needed to do that ?

Yes, just this:

```
dart run convenient_test_manager_dart # in one shell
flutter run path/to/your/integration_test/main_test.dart --host-vmservice-port 9753 --disable-service-auth-codes --dart-define CONVENIENT_TEST_APP_CODE_DIR=sth # in another shell, run your worker app
```

In one shell you run that convenient_test_manager_dart which is headless, i.e. a command line program.
In another shell you run that flutter test, i.e. worker. This is needed, just like in the traditional case you run `flutter test` and needs a simulator / real device.

**Comment by MCord** [comment]
oh I just figured out you are running `convenient_test_manager_dart` and not `convenient_test_manager`. it makes sense now. Silly me !

Thanks for the very fast response !

**Reply by fzyzcjy** [reply]
You are welcome!

**Comment by lehmannk** [comment]
Running the 2 commands above (in 2 independent terminals) then `convenient_test_manager_dart` terminates after the tests have completed but the app under test (started with `flutter run ....`) is still running. This blocks my CI integration. What am I missing here?

**Reply by fzyzcjy** [reply]
There are multiple possible ways. For example, you can kill the flutter process when convenient_test_manager_dart terminates. Or, you can modify your dart test script such that it terminates itself after all tests are done.
